cmd/nodecmd/destroy.go

```diff
@@ -126,14 +128,29 @@ func destroyNodes(_ *cobra.Command, args []string) error {
 		}
 		nodesToStop = append(nodesToStop, ltInstance)
 	}
-	awmRelayerHost, err := getAWMRelayerHost(clusterName)
+	nodeErrors := map[string]error{}
+	cloudSecurityGroupList, err := getCloudSecurityGroupList(nodesToStop)
 	if err != nil {
 		return err
 	}
-	nodeErrors := map[string]error{}
-	lastRegion := ""
-	var ec2Svc *awsAPI.AwsCloud
+	nodeToStopConfig, err := app.LoadClusterNodeConfig(nodesToStop[0])
+	if err != nil {
+		return err
+	}
+	// TODO: will need to change this logic if we decide to mix AWS and GCP instances in a cluster
+	filteredSGList := utils.Filter(cloudSecurityGroupList, func(sg regionSecurityGroup) bool { return sg.cloud == nodeToStopConfig.CloudService })
+	if len(filteredSGList) == 0 {
+		return fmt.Errorf("no endpoint found in the  %s", nodeToStopConfig.CloudService)
+	}
 	var gcpCloud *gcpAPI.GcpCloud
+	ec2SvcMap := make(map[string]*awsAPI.AwsCloud)
+	for _, sg := range filteredSGList {
+		sgEc2Svc, err := awsAPI.NewAwsCloud(awsProfile, sg.region)
```

Comment:
this assumes sg in filteredSGList are aws, so not so general (what happens if cluster is non AWS?)